src/test/java/net/snowflake/client/jdbc/ResultSetLatestIT.java

```diff
@@ -971,4 +979,138 @@ public void testLargeStringRetrieval() throws SQLException {
       fail("executeQuery should not fail");
     }
   }
+
+  // Test setting new connection property jdbc_arrow_treat_decimal_as_int=false. Connection property introduced after version 3.15.0.
+  @Test
+  public void testGetObjectForArrowResultFormatJDBCArrowDecimalAsIntFalse() throws SQLException {
+    Properties properties = new Properties();
+    properties.put("jdbc_arrow_treat_decimal_as_int", false);
+    try (Connection con = getConnection(properties);
+         Statement stmt = con.createStatement()) {
+      stmt.execute("alter session set jdbc_query_result_format = 'ARROW'");
+      stmt.execute(createTableSql);
+      stmt.execute(insertStmt);
+
+      // Test with jdbc_arrow_treat_decimal_as_int=false and JDBC_TREAT_DECIMAL_AS_INT=true
+      try (ResultSet rs = stmt.executeQuery(selectQuery)) {
+        while(rs.next()) {
+          assertEquals("class java.lang.Long", rs.getObject(1).getClass().toString());
```

Comment:
let's create a method `assertAllColumnsAreLongButBigIntIsBigDecimal` and use it in all the similar cases in added tests